File: common/src/test/java/dev/number6/slack/FakeHttpModule.java
package dev.number6.slack;

import dev.number6.slack.port.HttpPort;
import dagger.Module;
import dagger.Provides;

import javax.inject.Singleton;

@Module
public class FakeHttpModule {

    @Provides
    @Singleton
    public HttpPort providesHttp() {
        return new FakeHttpAdaptor();
    }
}

File: slack-reader/src/test/java/dev/number6/slackreader/dagger/FakeHttpModule.kt
package dev.number6.slackreader.dagger

import dagger.Module
import dagger.Provides
import dev.number6.slack.port.HttpPort
import javax.inject.Singleton

@Module
class FakeHttpModule {
    @Provides
    @Singleton
    fun providesHttp(): HttpPort {
        return FakeHttpAdaptor()
    }
}